### PR TITLE
Adds test for execution/mutable_state_builder.go

### PR DESCRIPTION
**Pull request by davidporter-id-au**
- Adds tests for GetCompletionEvent, GetStartEvent. 
- Adds a mock for shard.Context

This is part of a wider initiative to increase unit-test coverage for this package

### CODE REVIEW

File: service/history/shard/context.go [hunk 1]
```diff
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination interface_mock.go -package shard github.com/uber/cadence/history/shard/context Context
```
Comment (taylanisikdemir) [inline service/history/shard/context.go:21]: I'd rename interface_mock.go to context_mock.go for co-locating these and make relation obvious when looking at the file tree

File: service/history/execution/mutable_state_builder_test.go [hunk 2]
```diff
@@ -990,3 +994,274 @@ func (s *mutableStateSuite) buildWorkflowMutableState() *persistence.WorkflowMut
 		VersionHistories:    versionHistories,
 	}
 }
+
+func TestNewMutableStateBuilderWithEventV2(t *testing.T) {
+
```
Comment (taylanisikdemir) [inline service/history/execution/mutable_state_builder_test.go:999]: nit: remove empty first lines in functions

File: service/history/execution/mutable_state_builder_test.go [hunk 3]
```diff
@@ -990,3 +994,274 @@ func (s *mutableStateSuite) buildWorkflowMutableState() *persistence.WorkflowMut
...
+						ShardID:       common.IntPtr(shardID),
+						DomainName:    "domain",
+					}).Return(&persistence.ReadHistoryBranchResponse{
+					HistoryEvents: []*types.HistoryEvent{
+						exampleCompletionEvent,
+					},
+				}, nil)
+
+			},
+
```
Comment (taylanisikdemir) [inline service/history/execution/mutable_state_builder_test.go:1120]: nit: remove empty line

File: service/history/execution/mutable_state_builder_test.go [hunk 4]
```diff
@@ -990,3 +994,274 @@ func (s *mutableStateSuite) buildWorkflowMutableState() *persistence.WorkflowMut
...
+			},
+			historyManagerAffordance: func(historyManager *persistence.MockHistoryManager) {},
+			expectedResult:           nil,
+			expectedErr:              &types.InternalServiceError{Message: "unable to get workflow completion event"},
+		},
+	}
+
+	for name, td := range tests {
+		t.Run(name, func(t *testing.T) {
+
```
Comment (taylanisikdemir) [inline service/history/execution/mutable_state_builder_test.go:1174]: nit: remove empty first line of funcs

File: service/history/execution/mutable_state_builder_test.go [hunk 5]
```diff
@@ -990,3 +994,274 @@ func (s *mutableStateSuite) buildWorkflowMutableState() *persistence.WorkflowMut
...
+			domainCache := cache.NewMockDomainCache(ctrl)
+			domainCache.EXPECT().GetDomainName(gomock.Any()).Return("domain", nil).AnyTimes() // this isn't called on validation
+
+			td.currentState.eventsCache = events.NewCache(shardID, historyManager, config.NewForTest(), log.NewNoop(), metrics.NewNoopMetricsClient(), domainCache)
+			td.currentState.shard = shardContext
+
+			res, err := td.currentState.GetCompletionEvent(context.Background())
+
+			assert.Equal(t, td.expectedResult, res)
+			assert.Equal(t, td.expectedErr, err)
```
Comment (taylanisikdemir) [inline service/history/execution/mutable_state_builder_test.go:1191]: `assert.EqualError` does better job (I could be wrong. little experience with this assert package)

**Review by taylanisikdemir — APPROVED** [review]

File: service/history/execution/mutable_state_builder_test.go (hunk 2 above)
Comment (davidporter-id-au) [reply service/history/execution/mutable_state_builder_test.go:999]: sure, but I'd suggest putting that in lint if it's a thing we want to do more generally

File: service/history/execution/mutable_state_builder_test.go (hunk 5 above)
Comment (davidporter-id-au) [reply service/history/execution/mutable_state_builder_test.go]: I also admit not being super familiar with that specifically. It looks like it is just doing string comparison though so probably `assert.ErrorAs` and `assert.ErrorIs` I suspect. Let me look

File: service/history/execution/mutable_state_builder_test.go (hunk 2 above)
Comment (taylanisikdemir) [reply service/history/execution/mutable_state_builder_test.go:999]: Would be nice to remove generous new lines like this via linter which IMO hurts readability a little bit. Consistently spaced codebase is much easier to read thru. We are far from consistency so this new line is not in the top of my list :) Maybe @Groxx can quickly come up with the magic regex for this too

**Pull request merged**